build.gradle

```diff
@@ -404,6 +404,34 @@ allprojects {
   apply plugin: 'elasticsearch.formatting'
 }
 
+tasks.named("updateDaemonJvm") {
+  def myPlatforms = [
+    BuildPlatformFactory.of(
+      org.gradle.platform.Architecture.AARCH64,
+      org.gradle.platform.OperatingSystem.MAC_OS
+    ),
+    BuildPlatformFactory.of(
+      org.gradle.platform.Architecture.AARCH64,
+      org.gradle.platform.OperatingSystem.LINUX
+    ),
+    BuildPlatformFactory.of(
+      org.gradle.platform.Architecture.X86_64,
+      org.gradle.platform.OperatingSystem.LINUX
+    ),
+    BuildPlatformFactory.of(
+      org.gradle.platform.Architecture.X86_64,
+      org.gradle.platform.OperatingSystem.WINDOWS
+    ),
+    // anyone still using x86 osx?
+    BuildPlatformFactory.of(
+      org.gradle.platform.Architecture.X86_64,
+      org.gradle.platform.OperatingSystem.MAC_OS
+    )
+  ]
+  toolchainPlatforms.set(myPlatforms)
+  languageVersion = JavaLanguageVersion.of(21)
```

Comment:
@jozala brought up that those archived oracle jdks are not openjdk anymore and we might hit a license issue here. I wonder if we are okay here or if we should maybe use a different openjdk vendor. @mark-vieira do you have thoughts on this?

Reply:
I forget the exact license terms. We are "redistributing" so we're probably ok, but I guess it would be considered "commercial use". If there are any concerns I agree that just using Adoptium would be easiest. We're not picky here, since this is only used by build-time toolchains.